src/interactive-window/interactiveWindow.ts

```diff
@@ -186,22 +190,35 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
             try {
                 // Try creating a kernel
                 initializeInteractiveOrNotebookTelemetryBasedOnUserAction(this.owner, controller.connection);
-                const kernel = await connectToKernel(controller, this.serviceContainer, this.owner, notebook);
+
+                const updateMessage = (_: unknown, k: IKernel) => {
+                    this.updateSysInfoMessage(
+                        this.getSysInfoMessage(k.kernelConnectionMetadata, SysInfoReason.Start),
+                        false,
+                        sysInfoCell
+                    );
+                };
+                // When connecting, we need to update the sys info message
+                this.updateSysInfoMessage(
+                    this.getSysInfoMessage(controller.connection, SysInfoReason.Start),
+                    false,
+                    sysInfoCell
+                );
+                const kernel = await connectToKernel(
+                    controller,
+                    this.serviceContainer,
+                    this.owner,
+                    notebook,
+                    undefined,
+                    updateMessage
```

Comment:
Fixes https://github.com/microsoft/vscode-jupyter/issues/9454
If we pick another kernel as a result of ipykernel not being foundin current kernel, then we need to update the message that reads `connecting to kernel xyz`, this callback provides the ability to just that.